zip-proportional-output-fee-mechanism-pofm.rst

```diff
@@ -178,29 +134,63 @@ Open Issues
 > TODO: Remove this section once a decision is made.
 
 Possible alternatives for the parameters:
-* base_fee = 1000, marginal_fee = 250 in @nuttycom's proposal.
-* base_fee = 1000, marginal_fee = 1000 in @madars' proposal.
-* base_fee = 10000, marginal_fee = 2500 in @daira's proposal.
-* base_fee = 1000, marginal_fee = 1000 for Shielded, Shielding and De-shielding
-  transactions, and base_fee = 10000, marginal_fee = 10000 for Transparent transactions 
-  per @nighthawk24's proposal.
+
+* marginal_fee = 250 in @nuttycom's proposal.
+* marginal_fee = 1000 adapted from @madars' proposal.
+* marginal_fee = 2500 in @daira's proposal.
+* marginal_fee = 1000 for Shielded, Shielding and De-shielding
+  transactions, and marginal_fee = 10000 for Transparent transactions
+  adapted from @nighthawk24's proposal.
+
+(In @madars' and @nighthawk24's original proposals, there was an additional
+`base_fee` parameter that caused the relationship between fee and number
+of inputs/outputs to be non-proportional above the `grace_window_size`. This
+is no longer expressible with the formula specified above.)
+
+Logical actions
+'''''''''''''''
+
+The currently proposed formula may disadvantage Orchard transactions, as a
+result of an Orchard Action combining an input and an output. This means that
+Orchard effectively requires padding of either inputs or outputs to ensure that
+the number of inputs and outputs are the same. Usage of Sapling and transparent
+protocols does not require this padding, and so this could effectively
+discriminate against Orchard.
+
+An alternative formula, instead of using `inputs + outputs`, uses a concept
+of "logical actions", computed as follows::
```

Comment:
```suggestion
Rationale for logical actions
'''''''''''''''''''''''''''''

A previous proposal used `inputs + outputs` instead of logical actions.
This would have disadvantages Orchard transactions, as a result of an
Orchard Action combining an input and an output. The effect of this
combining is that Orchard requires padding of either inputs or outputs
to ensure that the number of inputs and outputs are the same. Usage of
Sapling and transparent protocols does not require this padding, and
so this could have effectively discriminated against Orchard.
```